### PR TITLE
Move to kebab case and add categories in model

### PR DESCRIPTION
**Pull request by maxandersen**
Why:

 * all other formatting in quarkus uses kebab-case
 * categories needed in registry

This change addreses the need by:

 * use kebab property naming for parsing of artifacts class.
 * rename shortName to short-name and put in metadata
 * add tests
 * add categories
 * move to use jackson parsing in preparation for move to yaml (I think we'll do that in a separate PR)

### CODE REVIEW

File: devtools/bom-descriptor-json/src/main/resources/extensions-overrides.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+
```
Comment (maxandersen) [inline devtools/bom-descriptor-json/src/main/resources/extensions-overrides.json]: Just draft example - will be removed before final review/merge.

**Comment by maxandersen** [comment]
@aloubyansky @ia3andy I would like to get review/confirmation on the general idea in this pr - majority is just updating the actual .json files; actual code is fairly small. It basically just clean up the descriptor structure and introduces notion of categories in platform descriptor. 

Once we merge this I'll do a follow up that moves the `quarkus-descriptor.json` to `quarkus-extension.yaml` - that one is a bit trickier because we have spread out reading of this file so would rather do that change in isolation.

File: independent-projects/bootstrap/maven-plugin/pom.xml [hunk 2]
```diff
@@ -1,74 +1,78 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
-    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (aloubyansky) [inline independent-projects/bootstrap/maven-plugin/pom.xml]: Could you please undo the formatting here?

File: independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java [hunk 3]
```diff
@@ -21,168 +21,235 @@
 import org.eclipse.aether.RepositorySystemSession;
 import org.eclipse.aether.repository.RemoteRepository;
 
-import com.eclipsesource.json.Json;
```
Comment (aloubyansky) [inline independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java]: Revert the formatting please. Or enable the Quarkus official one.

**Review by aloubyansky — CHANGES_REQUESTED** [review]

File: independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java (hunk 3 above)
Comment (maxandersen) [reply independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java]: seems like for some reason codeformatting is disabled on independent projects - when I run the codeformatter using ide-format the diff affects the following:

```
M core/src/main/java/io/quarkus/bootstrap/BootstrapClassLoaderFactory.java
 M core/src/main/java/io/quarkus/bootstrap/DefineClassVisibleURLClassLoader.java
 M core/src/main/java/io/quarkus/bootstrap/model/AppArtifactCoords.java
 M core/src/main/java/io/quarkus/bootstrap/model/AppArtifactKey.java
 M core/src/main/java/io/quarkus/bootstrap/model/AppDependency.java
 M core/src/main/java/io/quarkus/bootstrap/model/AppModel.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/AppModelResolver.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/BootstrapAppModelResolver.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapWagonProvider.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/BuildDependencyGraphVisitor.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/DeploymentInjectingDependencyVisitor.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenArtifactResolver.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenLocalRepositoryManager.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenModelBuilder.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenRepoInitializer.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/SecDispatcherImpl.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/SettingsDecrypterImpl.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalWorkspace.java
 M core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/ModelUtils.java
 M core/src/main/java/io/quarkus/bootstrap/util/BootstrapUtils.java
 M core/src/main/java/io/quarkus/bootstrap/util/IoUtils.java
 M core/src/main/java/io/quarkus/bootstrap/util/PropertyUtils.java
 M core/src/main/java/io/quarkus/bootstrap/util/ZipUtils.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/CollectDependenciesBase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/ResolverSetupCleanup.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/TsArtifact.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/TsDependency.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/TsJar.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/TsQuarkusExt.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/TsRepoBuilder.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/maven/test/MavenRepoInitializerTest.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/replace/test/SimpleReplacedDependencyTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/DependenciesOnDifferentVersionsOfAnArtifactTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/DependencyPresentTwiceInTheGraphInCompileAndProvidedScopesTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/DependencyPresentTwiceInTheGraphInCompileAndTestScopesTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/DependencyPresentTwiceInTheGraphTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/DependencyPresentTwiceInTheGraphWithDifferentClassifierAndTypeTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/DirectDependencyVersionOverridesTransitiveVersionTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/DirectOptionalOverridesTransitiveNonOptionalVersionTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/ExclusionsTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/OnlyDirectOptionalDepsAreCollectedTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/ProvidedScopeDepsAreNotCollectedTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/TestScopeIsNotAmongRuntimeDependenciesTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/resolver/test/TransitiveVersionOverridesTestCase.java
 M core/src/test/java/io/quarkus/bootstrap/util/ZipUtilsTest.java
 M core/src/test/java/io/quarkus/bootstrap/workspace/test/LocalWorkspaceDiscoveryTest.java
 M core/src/test/java/io/quarkus/bootstrap/workspace/test/MvnProjectBuilder.java
 M maven-plugin/src/main/java/io/quarkus/maven/AbstractTreeMojo.java
 M maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java
 M maven-plugin/src/test/java/io/quarkus/maven/TreeMojoTestBase.java
```

For this PR I've re-formatted just the single file you asked for but we should get it enabled for the project automatically going forward.

File: independent-projects/bootstrap/maven-plugin/pom.xml (hunk 2 above)
Comment (aloubyansky) [reply independent-projects/bootstrap/maven-plugin/pom.xml]: And this one?

File: independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java [hunk 4]
```diff
@@ -150,39 +167,88 @@ public void execute() throws MojoExecutionException {
...
+            metadata.set("short-name", extObject.get("shortName"));
+            extObject.remove("shortName");
+        }
+
+        extObject.set("metadata", metadata);
+
+        // TODO: remove before going to master
+        Path source = output
+                .resolve("../../../src/main/resources/META-INF/");
+        System.out.println("Try to save " + source);
```
Comment (aloubyansky) [inline independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java]: Forgot to remove?

File: independent-projects/tools/common/src/test/java/io/quarkus/test/platform/descriptor/loader/QuarkusTestPlatformDescriptorLoader.java [hunk 5]
```diff
@@ -139,7 +140,12 @@ public String getTemplate(String name) {
             @Override
             public <T> T loadResource(String name, ResourceInputStreamConsumer<T> consumer) throws IOException {
                 return loadStaticResource(name, consumer);
-            }};
+            }
+
```
Comment (aloubyansky) [inline independent-projects/tools/common/src/test/java/io/quarkus/test/platform/descriptor/loader/QuarkusTestPlatformDescriptorLoader.java]: Formatting issue.

File: independent-projects/tools/platform-descriptor-api/src/main/java/io/quarkus/dependencies/Category.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
+package io.quarkus.dependencies;
+
+import java.util.Map;
+
+public class Category {
```
Comment (aloubyansky) [inline independent-projects/tools/platform-descriptor-api/src/main/java/io/quarkus/dependencies/Category.java]: Looks like another non-matching formatting.

File: independent-projects/tools/platform-descriptor-api/src/main/java/io/quarkus/dependencies/Extension.java [hunk 7]
```diff
@@ -17,254 +18,280 @@
  */
 public class Extension {
 
-    private String artifactId;
```
Comment (aloubyansky) [inline independent-projects/tools/platform-descriptor-api/src/main/java/io/quarkus/dependencies/Extension.java]: Formatting.

File: independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java (hunk 4 above)
Comment (maxandersen) [reply independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java]: it’s the code generating the update json and now yaml. will be removed before final merge.

File: independent-projects/tools/common/src/test/java/io/quarkus/test/platform/descriptor/loader/QuarkusTestPlatformDescriptorLoader.java (hunk 5 above)
Comment (maxandersen) [reply independent-projects/tools/common/src/test/java/io/quarkus/test/platform/descriptor/loader/QuarkusTestPlatformDescriptorLoader.java]: See above comments - independent projects .java file were NOT formatted even before this PR.

File: independent-projects/tools/platform-descriptor-api/src/main/java/io/quarkus/dependencies/Extension.java (hunk 7 above)
Comment (maxandersen) [reply independent-projects/tools/platform-descriptor-api/src/main/java/io/quarkus/dependencies/Extension.java]: file now using quarkus format.

File: independent-projects/tools/platform-descriptor-api/src/main/java/io/quarkus/dependencies/Category.java (hunk 6 above)
Comment (maxandersen) [reply independent-projects/tools/platform-descriptor-api/src/main/java/io/quarkus/dependencies/Category.java]: file now using quarkus format.

File: independent-projects/tools/common/src/test/java/io/quarkus/test/platform/descriptor/loader/QuarkusTestPlatformDescriptorLoader.java (hunk 5 above)
Comment (maxandersen) [reply independent-projects/tools/common/src/test/java/io/quarkus/test/platform/descriptor/loader/QuarkusTestPlatformDescriptorLoader.java]: file now using quarkus format.

File: independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java (hunk 4 above)
Comment (maxandersen) [reply independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java]: removed it now so we can get this change into master - then from there on I can do a simpler yaml change seperatly.

**Comment by maxandersen** [comment]
Note: reason for the whitespace diff in ..json files is that we moved to use jackson for the read/write and it uses " : " rather than ": " as used by eclipsesource json. didn't find a flag to control that and since trying to move to yaml anyways I didn't want to spend unnecssary amount of time tweaking the pretty print of json.

File: independent-projects/tools/common/src/test/java/io/quarkus/test/platform/descriptor/loader/QuarkusTestPlatformDescriptorLoader.java (hunk 5 above)
Comment (maxandersen) [reply independent-projects/tools/common/src/test/java/io/quarkus/test/platform/descriptor/loader/QuarkusTestPlatformDescriptorLoader.java]: file now using quarkus format.

**Comment by maxandersen** [comment]
I got no idea why the builds are now failing with vm errors - this is similar kind of error I saw in master *before* I did the changes in this PR.

**Review by aloubyansky — APPROVED** [review]

**Comment by maxandersen** [comment]
now actually seeing relevant test errors....investigating!

**Comment by aloubyansky** [comment]
I approved before checking the CI. There are NPEs in the tests though.

**Comment by maxandersen** [comment]
> I approved before checking the CI. There are NPEs in the tests though.

yes - just saw them. looking.

**Comment by maxandersen** [comment]
ok, shortName failing null check - that was a change I did last so must have not done full IT test on that. fixing.

**Comment by maxandersen** [comment]
okey, that was easy fix - pushed - lets see what the tests says this time ;)

**Comment by maxandersen** [comment]
all test passed, but then noticed the dummy extensions-override.json was still there - so now removed but I expect the tests to still pass and we should push then.

but not going to wait for that to complete before I go sleep - so if someone sees this be green before I wake tomorrow then please merge ;)

**Review by ia3andy — APPROVED** [review]
LGTM very cool @maxandersen

**Pull request merged**